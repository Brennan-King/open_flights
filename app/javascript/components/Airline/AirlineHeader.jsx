import React from 'react';
import PropTypes from 'prop-types';
import './AirlineHeader.scss';

const propTypes = {
  /** Object containing an airline's attributes. */
  airlineAttributes: PropTypes.object,
};

const propDefaults = {
  airlineAttributes: {},
};

/**
 * Constructs an airline header component used to render a header
 * for a specific airline.
 * @param {object} props the props used to configure the airline header
 * @return {node} a div containing the configured airline header
 */
const AirlineHeader = (props) => {
  const {airlineAttributes} = props;
  const airlineName = airlineAttributes.name;
  const airlineImageUrl = airlineAttributes.image_url;
  const airlineAverageScore = airlineAttributes.avg_score;
  const totalAirlineReviews = airlineAttributes.reviews ?
    airlineAttributes.reviews.length : 0;
  const totalAirlineReviewsText = `${totalAirlineReviews} user reviews`;
  const airlineRatingOutOfFiveText = `${airlineAverageScore} out of 5`;

  return (
    <div className="airline-header-container">
      <h1 className="airline-header">
        <img
          className="airline-header-image"
          src={airlineImageUrl}
          alt={airlineName} />
        {airlineName}
      </h1>
      <div>
        <div className="airline-total-reviews">
          {airlineRatingOutOfFiveText}
        </div>
        <div className="airline-rating-out-of-five">
          {totalAirlineReviewsText}
        </div>
      </div>
    </div>
  );
};

AirlineHeader.propDefaults = propDefaults;
AirlineHeader.propTypes = propTypes;
export default AirlineHeader;
