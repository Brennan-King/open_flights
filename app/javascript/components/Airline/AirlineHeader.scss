.airline-header-container {
  padding: 50px 100px 50px 0;
  font-size: 30px;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.airline-header-image {
  height: 60px;
  width: 60px;
  border-radius: 100%;
  border: 1px solid darkgray;
  margin-bottom: -8px;
  margin-right: 5px;
}

.airline-total-reviews {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.airline-rating-out-of-five {
  font-size: 18px;
  font-weight: bold;
  padding: 10px 0;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}