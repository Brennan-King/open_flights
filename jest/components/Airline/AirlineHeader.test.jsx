/* eslint-disable max-len */
import 'jsdom-global/register';
import React from 'react';
import AirlineHeader from '../../../app/javascript/components/Airline/AirlineHeader';
import {configure, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({adapter: new Adapter()});

/**
 * Test suite for AirlineHeader component
 */
describe('AirlineHeader', () => {
  it('should render an AirlineHeader component with deafult props', () => {
    const component = mount(
        <AirlineHeader />,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render an AirlineHeader component with airline attributes', () => {
    const component = mount(
        <AirlineHeader
          airlineAttributes={
            {
              name: 'testAirlineName',
              image_url: 'testImageUrl',
              avg_score: 5,
              reviews: [],
            }
          }
        />,
    );

    expect(component).toMatchSnapshot();
  });
});
