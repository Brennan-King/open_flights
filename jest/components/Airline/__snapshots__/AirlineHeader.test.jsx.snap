// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirlineHeader should render an AirlineHeader component with airline attributes 1`] = `
<AirlineHeader
  airlineAttributes={
    Object {
      "avg_score": 5,
      "image_url": "testImageUrl",
      "name": "testAirlineName",
      "reviews": Array [],
    }
  }
>
  <div
    className="airline-header-container"
  >
    <h1>
      <img
        alt="testAirlineName"
        className="airline-header-image"
        src="testImageUrl"
      />
      testAirlineName
    </h1>
    <div>
      <div
        className="airline-total-reviews"
      >
        5 out of 5
      </div>
      <div
        className="airline-rating-out-of-five"
      >
        0 user reviews
      </div>
    </div>
  </div>
</AirlineHeader>
`;

exports[`AirlineHeader should render an AirlineHeader component with deafult props 1`] = `
<AirlineHeader
  airlineAttributes={
    Object {
      "avg_score": 0,
      "image_url": "",
      "name": "",
      "reviews": Array [],
    }
  }
>
  <div
    className="airline-header-container"
  >
    <h1>
      <img
        alt=""
        className="airline-header-image"
        src=""
      />
    </h1>
    <div>
      <div
        className="airline-total-reviews"
      >
        0 out of 5
      </div>
      <div
        className="airline-rating-out-of-five"
      >
        0 user reviews
      </div>
    </div>
  </div>
</AirlineHeader>
`;
